test/t8_schemes/t8_gtest_nca.cxx

```diff
@@ -151,165 +157,165 @@ TEST_P(nca, nca_check_deep){
+                     ts->t8_element_get_linear_id (desc_b, k), k);
         }
-        /* We reused parent_a, hence we have to recompute the correct parent */
-        ts->t8_element_parent(desc_a, parent_a);
+
+        t8_debugf ("id of the correct nca: %li, level: %i\n",
+                   ts->t8_element_get_linear_id (check_nca, level_c),
+                   level_c);
+
+        t8_debugf ("id of the computed nca: %li, level: %i\n",
+                   ts->t8_element_get_linear_id (check, level_nca),
+                   level_nca);
+
+        SC_ABORT ("Computed nca is not the correct nca!\n");
+      }
+      /* parent_a stays fixed, b-part goes one level deeper into the recursion */
+      t8_recursive_nca_check (check_nca, desc_a, parent_b, check, parent_a,
+                              desc_b, max_lvl, ts);
+      /* We reused parent_b, hence we have to recompute the correct parent */
+      ts->t8_element_parent (desc_b, parent_b);
     }
+    /* We reused parent_a, hence we have to recompute the correct parent */
+    ts->t8_element_parent (desc_a, parent_a);
+  }
 }
 
-
 /* Recursively check the computation of the nca. recursion_depth defines up to which
  * level we compute descendants of correct_nca that should have correct_nca as the 
  * output of t8_element_nca.*/
-TEST_P(nca, recursive_check)
+TEST_P (nca, recursive_check)
 {
 #ifdef T8_ENABLE_LESS_TESTS
-    const int recursion_depth = 3;
+  const int           recursion_depth = 3;
 #else
 /* User lower recursion depth for pyramids, it takes to much time otherwise */
-    const int recursion_depth = 4;
+  const int           recursion_depth = 4;
 #endif
-    t8_element_t *parent_a, *parent_b;
-    int     num_children;
-    num_children = ts->t8_element_num_children(correct_nca);
-    ts->t8_element_new (1, &parent_a);
-    ts->t8_element_new (1, &parent_b);
-    if(num_children > 1){
-        ts->t8_element_child(correct_nca, 0, parent_a);
-        ts->t8_element_child(correct_nca, 1, parent_b);
-        /* if you activate the following part, the testing takes around 30min per 3D class */
-        int     i,j;
-        for(i = 0; i < num_children-1; i++){
-            ts->t8_element_child(correct_nca, i, parent_a);
-            for(j = i+1; j < num_children; j++){
-                ts->t8_element_child(correct_nca, j, parent_b);
-                t8_recursive_nca_check(correct_nca, desc_a, desc_b, check, parent_a, parent_b, recursion_depth, ts);
-            }
-        }
-    }
-    else{
-        GTEST_SKIP();
+  t8_element_t       *parent_a, *parent_b;
+  int                 num_children;
+  num_children = ts->t8_element_num_children (correct_nca);
+  int                 i, j;
+  ts->t8_element_new (1, &parent_a);
+  ts->t8_element_new (1, &parent_b);
+  if (num_children > 1) {
+    ts->t8_element_child (correct_nca, 0, parent_a);
+    ts->t8_element_child (correct_nca, 1, parent_b);
+    for (i = 0; i < num_children - 1; i++) {
```

Comment:
My crude suggestion as we discussed. Something like this?
```
      for (k = 0; k < num_children; k++) {
        ...
        for (l = 0; l < num_children; l++) {
              if (k == l) continue;
              ...
        }
    }
```

Reply:
Does this test take 30 min to compute, or is it faster now? I assume ` recursion_depth = 3;` does not take that long.

Reply:
I don't know about the CI, but these are the runtimes on my machine, which are acceptable imo:
```
[ RUN      ] t8_gtest_nca/nca.recursive_check/1
[       OK ] t8_gtest_nca/nca.recursive_check/1 (0 ms)
[ RUN      ] t8_gtest_nca/nca.recursive_check/2
[       OK ] t8_gtest_nca/nca.recursive_check/2 (6 ms)
[ RUN      ] t8_gtest_nca/nca.recursive_check/3
[       OK ] t8_gtest_nca/nca.recursive_check/3 (5 ms)
[ RUN      ] t8_gtest_nca/nca.recursive_check/4
[       OK ] t8_gtest_nca/nca.recursive_check/4 (2045 ms)
[ RUN      ] t8_gtest_nca/nca.recursive_check/5
[       OK ] t8_gtest_nca/nca.recursive_check/5 (1567 ms)
[ RUN      ] t8_gtest_nca/nca.recursive_check/6
[       OK ] t8_gtest_nca/nca.recursive_check/6 (2110 ms)
[ RUN      ] t8_gtest_nca/nca.recursive_check/7
[       OK ] t8_gtest_nca/nca.recursive_check/7 (12540 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/0
[  SKIPPED ] t8_gtest_nca/nca.resursive_check_higher_level/0 (0 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/1
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/1 (0 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/2
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/2 (20 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/3
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/3 (33 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/4
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/4 (956 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/5
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/5 (1536 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/6
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/6 (2822 ms)
[ RUN      ] t8_gtest_nca/nca.resursive_check_higher_level/7
[       OK ] t8_gtest_nca/nca.resursive_check_higher_level/7 (5920 ms)
```